memory/alloydb/chat_message_history_test.go

```diff
@@ -0,0 +1,114 @@
+package alloydb_test
+
+import (
+	"context"
+	"github.com/tmc/langchaingo/internal/alloydbutil"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/memory/alloydb"
+	"os"
+	"testing"
+)
+
+type chatMsg struct{}
+
+func (c chatMsg) GetType() llms.ChatMessageType {
+	return llms.ChatMessageTypeHuman
+}
+
+func (c chatMsg) GetContent() string {
+	return "test content"
+}
+
+func getEnvVariables(t *testing.T) (string, string, string, string, string, string, string) {
+	t.Helper()
+
+	username := os.Getenv("ALLOYDB_USERNAME")
+	if username == "" {
+		t.Skip("ALLOYDB_USERNAME environment variable not set")
+	}
+	password := os.Getenv("ALLOYDB_PASSWORD")
+	if password == "" {
+		t.Skip("ALLOYDB_PASSWORD environment variable not set")
+	}
+	database := os.Getenv("ALLOYDB_DATABASE")
+	if database == "" {
+		t.Skip("ALLOYDB_DATABASE environment variable not set")
+	}
+	projectID := os.Getenv("ALLOYDB_PROJECT_ID")
+	if projectID == "" {
+		t.Skip("ALLOYDB_PROJECT_ID environment variable not set")
+	}
+	region := os.Getenv("ALLOYDB_REGION")
+	if region == "" {
+		t.Skip("ALLOYDB_REGION environment variable not set")
+	}
+	instance := os.Getenv("ALLOYDB_INSTANCE")
+	if instance == "" {
+		t.Skip("ALLOYDB_INSTANCE environment variable not set")
+	}
+	cluster := os.Getenv("ALLOYDB_CLUSTER")
+	if cluster == "" {
+		t.Skip("ALLOYDB_CLUSTER environment variable not set")
+	}
+
+	return username, password, database, projectID, region, instance, cluster
+}
+
+func setEngine(t *testing.T) (alloydbutil.PostgresEngine, error) {
+	username, password, database, projectID, region, instance, cluster := getEnvVariables(t)
+	ctx := context.Background()
+	pgEngine, err := alloydbutil.NewPostgresEngine(ctx,
+		alloydbutil.WithUser(username),
+		alloydbutil.WithPassword(password),
+		alloydbutil.WithDatabase(database),
+		alloydbutil.WithAlloyDBInstance(projectID, region, cluster, instance),
+	)
+	if err != nil {
+		t.Fatal("Could not set Engine: ", err)
+	}
+
+	return *pgEngine, nil
+}
```

Comment:
I think I left this comment on another PR, but I think this function shouldn't return an error AND require `testing.T` -- it should be one or the other (currently no error is returned).

So we should do one of the following to `setEngine`:
1. Refactor so that it doesn't use testing.T and returns and error
2. Refactor so that it doesn't return and error and uses testing.T

Reply:
just changed it to return the err, like we did on the previous PR! 
we do need the testing.T to get the env variables tho!